src/YoutubeAtom.tsx

```diff
@@ -311,12 +295,15 @@ export const YoutubeAtom = ({
 
             {(overlayImage || posterImage) && (
                 <div
-                    onClick={() => player?.playVideo()}
+                    onClick={() => {
+                        setHasUserLaunchedPlay(true);
+                        player.current?.playVideo();
+                    }}
                     onKeyDown={(e) => {
                         const spaceKey = 32;
                         const enterKey = 13;
                         if (e.keyCode === spaceKey || e.keyCode === enterKey)
-                            player?.playVideo();
+                            player.current?.playVideo();
```

Comment:
We don't set the play state here (as above) so this is a good argument for setting that `hasStartedPlaying` state in response to the listener rather than the click/keypress event